### PR TITLE
Honor charm channel in bundles

### LINKED ISSUE #496: Bundle deploy doesn't honor charm channels
The bundle handler's `AddCharmChange` implementation is [ignoring the channel](https://github.com/juju/python-libjuju/blob/d6fae9e/juju/bundle.py#L439) even though it [does have the channel available](https://github.com/juju/python-libjuju/blob/d6fae9e/juju/bundle.py#L447-L448).  E.g., a deploy logged this:

```
INFO     juju.bundle:bundle.py:166 Applying change: upload charm cs:~containers/kubernetes-worker for series focal from channel edge
```

But in fact deployed the charm rev from stable.

### PR DESCRIPTION
**Pull request by johnsca**
Fixes #496

### CODE REVIEW

File: juju/bundle.py [hunk 1]
```diff
@@ -436,7 +436,7 @@ async def run(self, context):
 
         entity_id = await context.charmstore.entityId(self.charm)
```
Comment (Cynerva) [inline juju/bundle.py:437]: I suspect the channel might need to be passed into charmstore.entityId too: https://github.com/juju/theblues/blob/a8cf043507cde71eba77727f2e6952d323349760/theblues/charmstore.py#L329

**Comment by SimonRichardson** [comment]
Also be aware this isn't always the case, it's a 2.7 feature I believe.

**Comment by johnsca** [comment]
@SimonRichardson I'm not sure what you mean.  Even the `ClientFacade.AddCharm` in `_client1.py` [supports the channel arg](https://github.com/juju/python-libjuju/blob/eebf2b3/juju/client/_client1.py#L4177).  If you mean that it's not supported in the bundle, then I think that it would be ignored or rejected by the `BundleFacade` (if that is even available there).

**Comment by SimonRichardson** [comment]
> If you mean that it's not supported in the bundle, then I think that it would be ignored or rejected by the `BundleFacade` (if that is even available there).

This ☝️

**Review by SimonRichardson — APPROVED** [review]

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**